consumer/sm_context.go

```diff
@@ -313,6 +313,10 @@ func SendUpdateSmContextXnHandover(
 	ue *amf_context.AmfUe, smContext *amf_context.SmContext, n2SmType models.N2SmInfoType, N2SmInfo []byte) (
 	*models.UpdateSmContextResponse, *models.UpdateSmContextErrorResponse, *models.ProblemDetails, error,
 ) {
+	// Check if the smContext is nil to prevent nil pointer dereference
+	if smContext == nil {
+		return nil, nil, nil, fmt.Errorf("smContext is nil")
+	}
```

Comment:
What was the outcome of the Xn handover after this fix?

Reply:
Before this fix, if the PDU session ID was invalid, the AMF would crash. However, after this fix, the AMF properly sends a Path Switch Request Failure to the gNodeB. You can verify this by checking the AMF logs and PCAPs.

[amf_logs_Xn_Handover_Invalid_PDU_Session_ID.zip](https://github.com/user-attachments/files/17108356/amf_logs_Xn_Handover_Invalid_PDU_Session_ID.zip)